jupyterlab_git/git.py

```diff
@@ -485,13 +486,22 @@ async def log(self, path, history_count=10, follow_path=None):
             return {"code": code, "command": " ".join(cmd), "message": my_error}
 
         result = []
-        if is_single_file:
-            # an extra newline get outputted when --numstat is used
-            my_output = my_output.replace("\n\n", "\n")
         line_array = my_output.splitlines()
-        i = 0
+
+        if is_single_file:
+            parsed_lines = []
+            for line in line_array:
+                if "\0" in line:
+                    split_line = line.replace("\0", "\t").split("\t")
+                    parsed_lines.append("\t".join(split_line[:-1]))
+                    if split_line[-1]:
+                        parsed_lines.append(split_line[-1])
+                else:
+                    parsed_lines.append(line)
```

Comment:
I think we can simplify this. The only case where `split_line[-1]` is empty is the last entry otherwise it will be the previous commit. So

```suggestion
            for line in line_array[:-1]:
                parsed_lines.extend(line.split("\0\0", maxsplit=1))
            parsed_lines.append(line_array[-1].strip("\0"))
```